### DISCUSSION TITLE
Library -> Hidden shows files, but "No pictures found" when clicked into

### DISCUSSION THREAD

**Discussion by vorezal** [photoprism/photoprism#3489] [Q&A]
I've run into a few issues with (slightly?) corrupt files when importing images that have been modified by other apps or services. I'm not certain of the root cause, but when these photos are processed they end up in the "Hidden" section of the library. That's all fine, however when I actually click into that section of the library I don't see any files displayed. No indication even of which files are impacted so I can find and fix or remove them. I've been doing a SQL query to find the impacted originals, open the images up in Gimp (which has yet to fail on me so far), re-export them, and then reindex the originals folder, which seems to work at least. Has anyone else seen the behavior where hidden files don't appear in the UI though? I wasn't able to find anything on this specific issue, though it's always possible my search skills are failing me.

I'm also going to do a test reimport of the original image, both by directly dropping it into the import folder and via WebDav and see what happens. I'll post logs and an example photo if I can reliably reproduce. I'm running 'latest' installed with docker compose on Windows 10 and Docker Desktop. And if this all pans out and I'm not just reporting something already known I can certainly write up a bug report.

**Comment by lastzero** [comment]
Would you be able to share details on the types of files and the version you are using?

**Reply by vorezal** [reply]
The files themselves are jpg and I am running PhotoPrism version 230615-90a18f6e7-Linux-AMD64-Plus. I was able to reproduce with a personal photo and these were the logs during the import (I changed the name of the phone to "device" though):

2023-06-19 15:06:45 INFO index: flagged 1 photo as hidden [54.774694ms]
2023-06-19 15:06:45 INFO index: updated 44 photos [1.820219511s]
2023-06-19 15:05:48 INFO import completed in 7 s
2023-06-19 15:05:42 INFO import: deleted empty folder 'device'
2023-06-19 15:05:42 INFO import: deleted empty folder 'device'
2023-06-19 15:05:42 INFO import: added main jpg file 2023/06/20230619_190216_8CDFD628.jpg
2023-06-19 15:05:42 INFO media: 20230619_190216_8CDFD628.jpg was taken at 2023-06-19 19:05:41.2019824 +0000 UTC (file mod time)
2023-06-19 15:05:41 INFO convert: 20230619_190216_8CDFD628.jpg.jpg created in 283.231725ms (convert)
2023-06-19 15:05:41 INFO convert: converting 20230619_190216_8CDFD628.jpg to 20230619_190216_8CDFD628.jpg.jpg (convert)
2023-06-19 15:05:41 INFO import: moving main jpg file 'device/Instagram/IMG_20230430_072807_298.jpg' to 2023/06/20230619_190216_8CDFD628.jpg
2023-06-19 15:05:41 INFO media: IMG_20230430_072807_298.jpg was taken at 2023-06-19 19:02:16.1918353 +0000 UTC (file mod time)
2023-06-19 15:05:40 INFO import: found no .ppignore file
2023-06-19 15:05:40 INFO copying files from import
2023-06-19 15:02:16 INFO webdav: PUT /import/device/Instagram/IMG_20230430_072807_298.jpg

Afterwards I see this in the UI:
![hidden](https://github.com/photoprism/photoprism/assets/37914382/36694407-9ecf-45ba-b364-3c58b2c6f964)
![hidden2](https://github.com/photoprism/photoprism/assets/37914382/1cac1e09-8483-42c0-90b5-4eadadce4d40)

I can send the photo itself if it would be helpful, though please let me know if there is a way to get it to you without posting publicly.

**Reply by lastzero** [reply]
Can you check if these are really JPEG files? Sometimes files have wrong extensions.

**Reply by vorezal** [reply]
Looks like the example I have is in fact an HEIC/HEIF file with a .jpg file extension. It includes "ftypheic" near the beginning of the file as text. Would that prevent it from showing up in the hidden files section?

**Reply by lastzero** [reply]
Yes. We also have users with JPEG files that have a HEIC extension. Google Photos sometimes does this.

**Reply by vorezal** [reply]
I understand this could cause a problem with format detection or conversion, and that seems completely reasonable and expected. I'm less concerned with why the file is being flagged as hidden and more with why, once it has been, I am unable to find it anywhere in the PhotoPrism UI.

**Reply by lastzero** [reply]
It's because there isn't even one file that could have been indexed as the main file is of the wrong type and the sidecar couldn't be created. We surely could display even better log messages for this case but didn't anticipate it and so didn't have the opportunity to do this yet.

**Reply by vorezal** [reply]
I see. Well, whatever you think is the best way to handle it. Ideally I'd like a way to get to the path and filename of the original so I know what to look at and try to fix. Logs is an option, though I do tend not to notice details there, especially if I'm importing hundreds or thousands of photos at once and the log event gets buried. Functionally it would be great if the hidden files screen could list files even if they completely failed indexing so it matches the count next to the menu button. As a workaround I can do it in SQL for now though.

Let me know if you would like me to write up a bug report, feature request, or some other kind of ticket for this unless you think it is covered by something already. Thank you for your time and help.

**Reply by lastzero** [reply]
As this doesn't seem to exist yet (or with a lower log level), you are welcome to add a feature request for showing an "invalid file type" warning in Library > Errors. These are recorded permanently, so you don't have to watch the live logs.
